webui/react/src/services/api-ts-sdk/api.ts

```diff
@@ -14597,11 +14603,12 @@ export const InternalApiFactory = function (configuration?: Configuration, fetch
          * @param {number} [paginationLimit] The amount of records limited in the results.
          * @param {string} [resourcePool] The target resource-pool for agent resource manager.
          * @param {'ORDER_BY_UNSPECIFIED' | 'ORDER_BY_ASC' | 'ORDER_BY_DESC'} [orderBy] Order results in either ascending or descending order by the number of jobs ahead.   - ORDER_BY_UNSPECIFIED: Returns records in no specific order.  - ORDER_BY_ASC: Returns records in ascending order.  - ORDER_BY_DESC: Returns records in descending order.
+         * @param {Array<'STATE_UNSPECIFIED' | 'STATE_QUEUED' | 'STATE_SCHEDULED' | 'STATE_SCHEDULED_BACKFILLED'>} [states] Filter to jobs with states among those given.   - STATE_UNSPECIFIED: Unspecified state.  - STATE_QUEUED: Job is queued and waiting to be schedlued.  - STATE_SCHEDULED: Job is scheduled.  - STATE_SCHEDULED_BACKFILLED: Job is scheduled as a backfill.
          * @param {*} [options] Override http request option.
          * @throws {RequiredError}
          */
-        getJobs(paginationOffset?: number, paginationLimit?: number, resourcePool?: string, orderBy?: 'ORDER_BY_UNSPECIFIED' | 'ORDER_BY_ASC' | 'ORDER_BY_DESC', options?: any) {
-            return InternalApiFp(configuration).getJobs(paginationOffset, paginationLimit, resourcePool, orderBy, options)(fetch, basePath);
+        getJobs(paginationOffset?: number, paginationLimit?: number, resourcePool?: string, orderBy?: 'ORDER_BY_UNSPECIFIED' | 'ORDER_BY_ASC' | 'ORDER_BY_DESC', states?: Array<'STATE_UNSPECIFIED' | 'STATE_QUEUED' | 'STATE_SCHEDULED' | 'STATE_SCHEDULED_BACKFILLED'>, options?: any) {
+            return InternalApiFp(configuration).getJobs(paginationOffset, paginationLimit, resourcePool, orderBy, states, options)(fetch, basePath);
```

Comment:
not sure why the bindings get generated as 
```
states?: Array<'STATE_UNSPECIFIED' | 'STATE_QUEUED' | 'STATE_SCHEDULED' | 'STATE_SCHEDULED_BACKFILLED'>
```

even though the thing its generated from is also present in the bindings as `determinedv1JobState`. do you know if there is a way to make it just use generated struct here cc @hamidzr ?

this is the reason for the hacky decoder code below.

Reply:
yeah this is unfortunate but how the generator generates the parameters. it just behaes weird around enums and these I think come through in a sub optimal way through the generated open api spec as well. we added a hacky solution in our in house python generator to address this https://github.com/hamidzr/determined/blob/0d4fe2385f0777b650e8841cfce9c21c1c9276e1/bindings/generate_bindings.py#L506-L511